OptimizelySDKCore/OptimizelySDKCore/OPTLYHTTPRequestManager.m

```diff
@@ -374,8 +390,9 @@ - (dispatch_time_t)backoffDelay:(NSInteger)backoffRetryAttempt
     
     OPTLYLogDebug(OPTLYHTTPRequestManagerBackoffRetryStates, backoffRetryAttempt, exponentialMultiplier, delay_ns, delayTime);
     
-    // TODO: Wrap this in a TEST preprocessor definition
-    self.delaysTest[backoffRetryAttempt] = [NSNumber numberWithLongLong:delay_ns];
+    if (self.isRunningTest == YES) {
+        self.delaysTest[backoffRetryAttempt] = [NSNumber numberWithLongLong:delay_ns];
```

Comment:
I think you should still have a test to see if backOffRetryAttempt is larger than the size of delaysTest (just as a sanity check)